tensorboard/webapp/core/views/page_title_container.ts

```diff
@@ -36,12 +58,44 @@ import {State} from '../state';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class PageTitleContainer {
-  readonly title$ = this.store.pipe(
-    select(getEnvironment),
-    map((env) => env.window_title || 'TensorBoard'),
-    // (it's an empty string when the `--window_title` flag is not set)
+  private readonly getExperimentId$ = this.store.select(getRouteKind).pipe(
+    filter((routeKind) => routeKind === RouteKind.EXPERIMENT),
+    withLatestFrom(this.store.select(getExperimentIdsFromRoute)),
+    map(([, experimentIds]) =>
+      experimentIds && experimentIds.length === 1 ? experimentIds[0] : null
+    )
+  );
+
+  private readonly experiment$ = this.getExperimentId$.pipe(
+    filter(Boolean),
+    mergeMap((experimentId) => {
+      return this.store.select(getExperiment, {experimentId});
+    })
+  );
+
+  private readonly experimentName$ = this.experiment$.pipe(
+    map((experiment) => (experiment ? experiment.name : null))
+  );
+
+  readonly title$ = this.store.select(getEnvironment).pipe(
+    combineLatestWith(this.experimentName$),
+    map(([env, experimentName]) => {
+      if (env.window_title && env.window_title.length > 0) {
```

Comment:
We can drop the `&& env.window_title.length > 0` and the `&& experimentName.length > 0` on L86, since an empty JS string will be falsy.

Reply:
Removed the unnecessary checks.